### PR TITLE
F/wpc-482-publication-date-without-seconds

### PR DESCRIPTION
**Pull request by camilleory**
- New date formatting using moment.js, translatable
- Code updated in articleLists, pageList, publishPanel and tags

### CODE REVIEW

File: packages/editor/src/client/locales/en.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
         "addTag": "Press enter to add tag",
         "allowPeerPublishing": "Allow peers to publish this article.",
         "article": "Article",
-        "articlePending": "There is already a pending publication scheduled at {{pendingPublishDate, MM/DD/YYYY HH:mm}}. Publishing again will override that publication. ",
```
Comment (roosnic1) [inline packages/editor/src/client/locales/en.json:24]: i18next already has a date and time formatting mechanism built in. You see [here](https://www.i18next.com/translation-function/formatting) which formatting is allowed and then one you want is [moment.js](https://momentjs.com/docs/#/displaying/format/). This allows you to pass date object to the `t(...)` and you can define in the translation files how that date should be displayed.

File: packages/editor/src/client/panel/publishArticlePanel.tsx [hunk 2]
```diff
@@ -31,6 +31,9 @@ export function PublishArticlePanel({
 
   const {t} = useTranslation()
 
+  const testDate = new Date()
+  console.log(testDate)
+
```
Comment (roosnic1) [inline packages/editor/src/client/panel/publishArticlePanel.tsx:36]: I guess you don't need that anymore 😊

File: packages/editor/src/client/panel/publishArticlePanel.tsx [hunk 3]
```diff
@@ -41,7 +44,10 @@ export function PublishArticlePanel({
         {pendingPublishDate && (
           <Message
             type="warning"
-            description={t('articleEditor.panels.articlePending', {pendingPublishDate})}
+            description={t('articleEditor.panels.articlePending', {
```
Comment (roosnic1) [inline packages/editor/src/client/panel/publishArticlePanel.tsx:47]: With the formatting options in i18next it should be possible to pass just one date object and do the formatting in the translation files.

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 4]
```diff
@@ -194,15 +194,19 @@ export function ArticleEditor({id}: ArticleEditorProps) {
       setTagTitle(
         t('articleEditor.overview.pending', {
           date: new Date(articleData?.article?.pending?.publishAt ?? '').toDateString(),
-          time: new Date(articleData?.article?.pending?.publishAt ?? '').toLocaleTimeString()
+          time: new Date(articleData?.article?.pending?.publishAt ?? '')
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleEditor.tsx:197]: With the formatting options in i18next it should be possible to pass just one date object and do the formatting in the translation files.

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 5]
```diff
@@ -194,15 +194,19 @@ export function ArticleEditor({id}: ArticleEditorProps) {
...
+            .slice(0, -3)
         })
       )
     } else if (articleData?.article?.published) {
       setStateColor(StateColor.published)
       setTagTitle(
         t('articleEditor.overview.published', {
           date: new Date(articleData?.article?.published?.publishedAt ?? '').toDateString(),
-          time: new Date(articleData?.article?.published?.publishedAt ?? '').toLocaleTimeString()
+          time: new Date(articleData?.article?.published?.publishedAt ?? '')
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleEditor.tsx:207]: With the formatting options in i18next it should be possible to pass just one date object and do the formatting in the translation files.

File: packages/editor/src/client/routes/articleList.tsx [hunk 6]
```diff
@@ -143,13 +148,11 @@ export function ArticleList() {
             <Cell dataKey="published">
               {(articleRef: ArticleRefFragment) =>
                 articleRef.published?.publishedAt
-                  ? `${new Date(articleRef.published.publishedAt).toDateString()} ${new Date(
-                      articleRef.published.publishedAt
-                    ).toLocaleTimeString()}`
+                  ? `${dateTimeWithoutSecondsLocalString(
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleList.tsx:151]: I recommend to also use the i18next translation service to format the date. Might be a bit more work if it's just the date but it's cleaner this way.

File: packages/editor/src/client/routes/articleList.tsx [hunk 7]
```diff
@@ -143,13 +148,11 @@ export function ArticleList() {
...
+                      new Date(articleRef.published.publishedAt)
+                    )}`
                   : articleRef.pending?.publishAt
-                  ? `${new Date(articleRef.pending.publishAt).toDateString()} ${new Date(
-                      articleRef.pending.publishAt
-                    ).toLocaleTimeString()}`
+                  ? `${dateTimeWithoutSecondsLocalString(new Date(articleRef.pending?.publishAt))}`
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleList.tsx:155]: I recommend to also use the i18next translation service to format the date. Might be a bit more work if it's just the date but it's cleaner this way.

File: packages/editor/src/client/routes/articleList.tsx [hunk 8]
```diff
@@ -158,9 +161,7 @@ export function ArticleList() {
             <HeaderCell>{t('articles.overview.updated')}</HeaderCell>
             <Cell dataKey="modifiedAt">
               {({modifiedAt}: ArticleRefFragment) =>
-                `${new Date(modifiedAt).toDateString()} ${new Date(
-                  modifiedAt
-                ).toLocaleTimeString()}`
+                `${dateTimeWithSecondsLocalString(new Date(modifiedAt))}`
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleList.tsx:164]: I recommend to also use the i18next translation service to format the date. Might be a bit more work if it's just the date but it's cleaner this way.

File: packages/editor/src/client/routes/articleList.tsx [hunk 9]
```diff
@@ -305,17 +306,18 @@ export function ArticleList() {
             )}
 
             <DescriptionListItem label={t('articles.panels.createdAt')}>
-              {currentArticle?.createdAt && new Date(currentArticle.createdAt).toLocaleString()}
+              {currentArticle?.createdAt &&
+                dateTimeWithoutSecondsLocalString(new Date(currentArticle.createdAt))}
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleList.tsx:310]: I recommend to also use the i18next translation service to format the date. Might be a bit more work if it's just the date but it's cleaner this way.

File: packages/editor/src/client/routes/articleList.tsx [hunk 10]
```diff
@@ -305,17 +306,18 @@ export function ArticleList() {
...
             </DescriptionListItem>
 
             <DescriptionListItem label={t('articles.panels.updatedAt')}>
               {currentArticle?.latest.updatedAt &&
-                new Date(currentArticle.latest.updatedAt).toLocaleString()}
+                dateTimeWithSecondsLocalString(new Date(currentArticle.latest.updatedAt))}
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleList.tsx:315]: I recommend to also use the i18next translation service to format the date. Might be a bit more work if it's just the date but it's cleaner this way.

File: packages/editor/src/client/routes/articleList.tsx [hunk 11]
```diff
@@ -305,17 +306,18 @@ export function ArticleList() {
...
             </DescriptionListItem>
 
             {currentArticle?.latest.publishedAt && (
               <DescriptionListItem label={t('articles.panels.publishedAt')}>
-                {new Date(currentArticle.createdAt).toLocaleString()}
+                {dateTimeWithoutSecondsLocalString(new Date(currentArticle.createdAt))}
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleList.tsx:320]: I recommend to also use the i18next translation service to format the date. Might be a bit more work if it's just the date but it's cleaner this way.

File: packages/editor/src/client/routes/pageList.tsx [hunk 12]
```diff
@@ -95,6 +100,7 @@ export function PageList() {
     }
   }, [data?.pages])
 
+  console.log('new format- ' + dateTimeWithoutSecondsLocalString(new Date()))
```
Comment (roosnic1) [inline packages/editor/src/client/routes/pageList.tsx:103]: You can remove this also.

File: packages/editor/src/client/routes/pageList.tsx [hunk 13]
```diff
@@ -140,13 +146,9 @@ export function PageList() {
             <Cell dataKey="published">
               {(pageRef: PageRefFragment) =>
                 pageRef.published?.publishedAt
-                  ? `${new Date(pageRef.published.publishedAt).toDateString()} ${new Date(
-                      pageRef.published.publishedAt
-                    ).toLocaleTimeString()}`
+                  ? `${dateTimeWithoutSecondsLocalString(new Date(pageRef.published.publishedAt))}`
```
Comment (roosnic1) [inline packages/editor/src/client/routes/pageList.tsx:149]: I recommend to also use the i18next translation service to format the date. Might be a bit more work if it's just the date but it's cleaner this way.

File: packages/editor/src/client/routes/pageList.tsx [hunk 14]
```diff
@@ -140,13 +146,9 @@ export function PageList() {
...
                   : pageRef.pending?.publishAt
-                  ? `${new Date(pageRef.pending.publishAt).toDateString()} ${new Date(
-                      pageRef.pending.publishAt
-                    ).toLocaleTimeString()}`
+                  ? `${dateTimeWithoutSecondsLocalString(new Date(pageRef.pending?.publishAt))}`
```
Comment (roosnic1) [inline packages/editor/src/client/routes/pageList.tsx:151]: I recommend to also use the i18next translation service to format the date. Might be a bit more work if it's just the date but it's cleaner this way.

File: packages/editor/src/client/routes/pageList.tsx [hunk 15]
```diff
@@ -155,9 +157,7 @@ export function PageList() {
             <HeaderCell>{t('pages.overview.updated')}</HeaderCell>
             <Cell dataKey="modifiedAt">
               {({modifiedAt}: PageRefFragment) =>
-                `${new Date(modifiedAt).toDateString()} ${new Date(
-                  modifiedAt
-                ).toLocaleTimeString()}`
+                `${dateTimeWithSecondsLocalString(new Date(modifiedAt))}`
```
Comment (roosnic1) [inline packages/editor/src/client/routes/pageList.tsx:160]: I recommend to also use the i18next translation service to format the date. Might be a bit more work if it's just the date but it's cleaner this way.

File: packages/editor/src/client/utility.ts [hunk 16]
```diff
@@ -64,6 +64,31 @@ export function dateTimeLocalString(date: Date) {
   return `${year}-${month}-${day}T${hours}:${minutes}:${seconds}`
 }
 
+export function dateTimeWithSecondsLocalString(date: Date) {
```
Comment (roosnic1) [inline packages/editor/src/client/utility.ts:67]: Nice way of handling this but the formatting libraries already do this and are a bit more flexible :)

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: packages/editor/src/client/locales/en.json (hunk 1 above)
Comment (camilleory) [reply packages/editor/src/client/locales/en.json:24]: I couldn't make the formatting work directly in the translation file so I used the same method that had been used just above on line 14 and 16 (and some other places). It also allows the use of **at** and **on** between date and time. 
Are you saying the date should get formatted for all occurences only in this translation file ?

**Review by bishoymelek — APPROVED** [review]
🎉

File: packages/editor/src/client/locales/en.json (hunk 1 above)
Comment (camilleory) [reply packages/editor/src/client/locales/en.json:24]: I changed the formatting for the built-in option.

**Review by roosnic1 — APPROVED** [review]
👍🏾

**Pull request merged**